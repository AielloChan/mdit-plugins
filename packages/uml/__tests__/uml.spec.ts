import MarkdownIt from "markdown-it";
import { describe, expect, it } from "vitest";

import { uml } from "../src/index.js";

describe("uml", () => {
  it("should render without options", () => {
    const markdownIt = MarkdownIt({ linkify: true }).use(uml);

    expect(
      markdownIt.render(`
@start

abc

@end
    `),
    ).toMatchSnapshot();
  });

  it("should keep content as is", () => {
    const markdownIt = MarkdownIt({ linkify: true }).use(uml);

    expect(
      markdownIt.render(`
@start

Text with **bold** and \`code\`.

@end
    `),
    ).toMatchSnapshot();
  });

  it("should render with options", () => {
    const markdownIt = MarkdownIt({ linkify: true }).use(uml, {
      name: "test",
      open: "teststart",
      close: "testend",
      render: (tokens, index): string => {
        const token = tokens[index];
        const { content, info, type } = token;

        return `<Test class="${type}" title="${info}">${content}</Test>`;
      },
    });

    expect(
      markdownIt.render(`
@teststart

abc

def

@testend
`),
    ).toMatchSnapshot();
  });
});
