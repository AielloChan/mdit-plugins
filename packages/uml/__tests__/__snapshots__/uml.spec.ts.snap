// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`uml > should keep content as is 1`] = `
"<div class="uml" title="">
Text with **bold** and \`code\`.
</div>"
`;

exports[`uml > should render with options 1`] = `
"<Test class="test" title="">
abc

def
</Test>"
`;

exports[`uml > should render without options 1`] = `
"<div class="uml" title="">
abc
</div>"
`;
